.github/workflows/ci.yml

```
@@ -125,26 +125,66 @@ jobs:
           name: codecov-umbrella
           fail_ci_if_error: false
           token: ${{ secrets.CODECOV_TOKEN }}
-      # Coveralls is disabled since it is just annoying for parallel builds, see, e.g.,
+      # The standard setup of Coveralls is just annoying for parallel builds, see, e.g.,
       # https://github.com/trixi-framework/Trixi.jl/issues/691
       # https://github.com/coverallsapp/github-action/issues/47
       # https://github.com/coverallsapp/github-action/issues/67
-      # - uses: coverallsapp/github-action@master
+      # This standard setup is reproduced below for completeness.
       #   with:
       #     github-token: ${{ secrets.GITHUB_TOKEN }}
       #     flag-name: run-${{ matrix.trixi_test }}-${{ matrix.os }}-${{ matrix.version }}-${{ matrix.arch }}-${{ github.run_id }}
       #     parallel: true
       #     path-to-lcov: ./lcov.info
+      # Instead, we use a more tedious approach:
+      # - Store all inidvidual coverage files as artifacts (directly below)
+      # - Download and merge individual coverage reports in another step
+      # - Upload only the merged coverage report to Coveralls
+      - shell: bash
+        run: |
+          cp ./lcov.info ./lcov-${{ matrix.trixi_test }}-${{ matrix.os }}-${{ matrix.version }}-${{ matrix.arch }}.info
+      - uses: actions/upload-artifact@v2
+        with:
+          name: lcov-${{ matrix.trixi_test }}-${{ matrix.os }}-${{ matrix.version }}-${{ matrix.arch }}
+          path: ./lcov-${{ matrix.trixi_test }}-${{ matrix.os }}-${{ matrix.version }}-${{ matrix.arch }}.info
 
   finish:
     needs: test
     runs-on: ubuntu-latest
     steps:
-      # Coveralls is disabled since it is just annoying for parallel builds,
-      # see https://github.com/trixi-framework/Trixi.jl/issues/691
+      # The standard setup of Coveralls is just annoying for parallel builds, see, e.g.,
+      # https://github.com/trixi-framework/Trixi.jl/issues/691
+      # https://github.com/coverallsapp/github-action/issues/47
+      # https://github.com/coverallsapp/github-action/issues/67
+      # This standard setup is reproduced below for completeness.
       # - name: Coveralls Finished
       #   uses: coverallsapp/github-action@master
       #   with:
       #     github-token: ${{ secrets.GITHUB_TOKEN }}
       #     parallel-finished: true
+      # Instead, we use the more tedious approach described above.
+      # At first, we download th source files and all artifacts 
```

```suggestion
      # At first, we check out the repository and download all artifacts 
```
IMHO a little clearer